components/raftstore/src/store/async_io/write.rs

```diff
@@ -706,10 +722,8 @@ where
                 write_kv_time = duration_to_sec(now.saturating_elapsed());
                 STORE_WRITE_KVDB_DURATION_HISTOGRAM.observe(write_kv_time);
             }
-
-            self.batch.after_write_to_kv_db(&self.metrics);
         }
-
+        self.batch.after_write_to_kv_db(&self.metrics);
```

Comment:
If there is no write to kv db, then it's wrong to call `after_write_to_kv_db`.

Reply:
Understand it, I wiil execute it alone not in after_write_to_kv_db.